### PR TITLE
refac: Lower cyclomatic complexity and enhance maintainability of `calculation_core.py`

### PR DESCRIPTION
**Pull request by BjarkeMeier**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/package/calculation/calculation_core.py [hunk 1]
```diff
@@ -166,3 +159,82 @@ def execute(
...
+            args.calculation_grid_areas, all_metering_point_periods
+        )
+
+        metering_point_periods_df_without_grid_loss = (
+            prepared_data_reader.get_metering_point_periods_without_grid_loss(
+                all_metering_point_periods
+            )
+        )
+
+        grid_loss_metering_points_df = get_grid_loss_metering_points(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/calculation_core.py:188]: move down to basis data part?

File: source/databricks/calculation_engine/package/calculation/calculation_core.py [hunk 2]
```diff
@@ -108,53 +117,37 @@ def execute(
...
+            Colname.metering_point_id
+        ).distinct()
+
+        input_charges = prepared_data_reader.get_input_charges(
+            args.calculation_period_start_datetime,
+            args.calculation_period_end_datetime,
+            metering_point_period_ids,
+        )
+
+        metering_point_periods_for_basis_data = all_metering_point_periods
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/calculation_core.py:132]: move to basis data part?

File: source/databricks/calculation_engine/package/calculation/calculation_core.py [hunk 3]
```diff
@@ -45,47 +47,54 @@ def execute(
...
+        calculation_output = CalculationOutput()
 
-            if is_wholesale_calculation_type(args.calculation_type):
-                all_metering_point_periods = (
-                    get_metering_points_periods_for_wholesale_basis_data(
-                        all_metering_point_periods
-                    )
-                )
+        # cache of metering_point_time_series had no effect on performance (01-12-2023)
+        all_metering_point_periods = prepared_data_reader.get_metering_point_periods_df(
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/calculation_core.py:64]: could this function call and the one in line 70 be merged into one function call?

File: source/databricks/calculation_engine/package/calculation/calculation_core.py [hunk 4]
```diff
@@ -45,47 +47,54 @@ def execute(
...
+            grid_loss_responsible_df
+        )
 
-            metering_point_time_series = (
-                prepared_data_reader.get_metering_point_time_series(
-                    args.calculation_period_start_datetime,
-                    args.calculation_period_end_datetime,
-                    metering_point_periods_df_without_grid_loss,
-                )
+        metering_point_time_series = (
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/package/calculation/calculation_core.py:90]: would it help readibilty if the method in line 80 is absorbed into this call. For instance having one method, something like this:
`
    metering_point_time_series_for_calculation = prepared_data_reader.get_metering_point_time_series(
                args.calculation_period_start_datetime,
                args.calculation_period_end_datetime,
                all_metering_point_periods,
            )
`

File: source/databricks/calculation_engine/package/calculation/calculation_core.py (hunk 2 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/calculation/calculation_core.py:132]: It's used in line 135.

**Review by JesperGomes — APPROVED** [review]

File: source/databricks/calculation_engine/package/calculation/calculation_core.py (hunk 3 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/calculation/calculation_core.py:64]: Yes, I'll do more refactorings in an upcoming PR.

File: source/databricks/calculation_engine/package/calculation/calculation_core.py (hunk 4 above)
Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/calculation/calculation_core.py:90]: Yes, I'll do more refactorings in an upcoming PR.

**Pull request merged**